src/sagemaker/workflow/emr_step.py

```diff
@@ -47,6 +49,7 @@ def __init__(
         self.args = args
         self.main_class = main_class
         self.properties = properties
+        self.validate_prop_dicts()
```

Comment:
Nit: this method is not used else where so its name should follow the convention: `validate_prop_dicts`

Reply:
Sorry, just realized that the underscore I added in previous comment was swallowed
I meant: `self._validate_prop_dicts()`

Reply:
Oh yea, for self methods, good call.